core/src/themes/ionic/ionic.mixins.scss

```
@@ -41,6 +41,40 @@
   pointer-events: none;
 }
 
+// Focused mixin to be used on regular elements for the Ionic Theme focused state
+//
+// ex: :host(.ion-focused) .toggle-icon {
+//       @include globals.focused-state();
+//      }
+//
+// --------------------------------------------------
+@mixin focused-state(
+  $width: tokens.$ionic-border-size-050,
+  $style: tokens.$ionic-border-style-solid,
+  $color: tokens.$ionic-state-focus-1,
+  $addOffset: true
+) {
+  // If a user passes in null, then they are opting out
+  // of using a specific parameter and we should use the default
+  @if $width == null {
+    $width: tokens.$ionic-border-size-050;
+  }
+
+  @if $style == null {
+    $style: tokens.$ionic-border-style-solid;
+  }
+
+  @if $color == null {
+    $color: tokens.$ionic-state-focus-1;
+  }
+
+  @if $addOffset {
+    outline-offset: $width;
+  }
+
+  outline: $width $style $color;
```

```suggestion
  // Use default values if null is passed
  $width: if($width != null, $width, tokens.$ionic-border-size-050);
  $style: if($style != null, $style, tokens.$ionic-border-style-solid);
  $color: if($color != null, $color, tokens.$ionic-state-focus-1);

  outline: $width $style $color;

  @if $addOffset {
    outline-offset: $width;
  }
```

Simplified a bit: [Playground](https://sass-lang.com/playground/#eJx9UctOwzAQvPsrFqmHVor7QMAhFVL/gBMfkMbrspJjV7ENBdR/x9mYJrSlJ6+9M7Mz69VqQs5SLbeuVdhKT18ol4/LEu73h7U4a4ZPg9I7Q6oEPk4IH6qAUrs6erkqocXUEpuGDmSBX1H1mKkAmHyQCm8lXB9ddAgedYEY5jOodsa1J9AfC9yvlHrR2mMoIbQRxQy+0/NiAa8eQaGuognwXpmIHkiDjcYAedhXPtkd+SQ97Uu4e2ZUkVvFPxFm61GGjs3liM33C/YQrxfI+ToBLkcCfC+uRk/cRHYxGLLdCnvn2UJPZMQmRR5WxKs50aTLe+vZnZujOAoxT/Nk/k+Y5xV65m7I1iYqPPvu5OaC15D3ZHeyIaWSpxvsh/2hyJG36ZuuienK+F+/t6SeBqldi2gLYGYn+QPFtfz9)